packages/shared/src/components/RankProgress.tsx

```diff
@@ -23,6 +24,63 @@ import styles from './RankProgress.module.css';
 
 const notificationDuration = 300;
 
+interface GetRankElementProps {
+  shownRank: number;
+  rank: number;
+  badgeRef: MutableRefObject<SVGSVGElement>;
+  plusRef: MutableRefObject<HTMLElement>;
+  smallVersion: boolean;
+  showRankAnimation: boolean;
+  animatingProgress: boolean;
+}
+
+type RankElementProps = Pick<
+  GetRankElementProps,
+  'shownRank' | 'rank' | 'badgeRef'
+>;
+const RankElement = ({
+  shownRank,
+  rank,
+  badgeRef,
+}: RankElementProps): ReactElement => (
+  <Rank
+    rank={shownRank}
+    className={classNames(
+      'absolute inset-0 w-2/3 h-2/3 m-auto',
+      styles.rank,
+      shownRank && styles.hasRank,
+    )}
+    ref={badgeRef}
+    colorByRank={rank > 0}
+  />
+);
+
+const getRankElement = ({
+  shownRank,
+  rank,
+  badgeRef,
+  plusRef,
+  smallVersion,
+  showRankAnimation,
+  animatingProgress,
+}: GetRankElementProps): ReactElement => {
+  if (!smallVersion)
+    return (
+      <RankElement shownRank={shownRank} rank={rank} badgeRef={badgeRef} />
+    );
+
+  return !showRankAnimation || !animatingProgress ? (
+    <RankElement shownRank={shownRank} rank={rank} badgeRef={badgeRef} />
+  ) : (
+    <strong
+      ref={plusRef}
```

Comment:
Why did we have to have two refs to utilize? We are rendering conditionally so there should only be one reference in any scenario.

Reply:
Only one of them will render.
It just means the top one will never animate between the +1 and the rank, the bottom one can.

Reply:
Oh you mean why the badge and plus are different, the one is a SVG reference the other a HTML element reference.

Reply:
Sorry for the short unclear question. I mean why do we have to use two refs, namely `badgeRef` and `plusRef`, I think we can make them use just one ref since only one element will be available in every scenario.

Reply:
Not sure if it's possible.
The one is a SVG the other a HTML element.

Once the current ref is set to SVG you can't swap over to HTML.

Reply:
I think it should be possible through callback ref but, I can say it might not be worth the change. Well done! 💪